### ISSUE TITLE
Personal Desk Booking Search Function

### ISSUE THREAD

**Issue by qy21gafy** [qy21gafy/AMOSSharedDeskFeatureBoard#13]
## User story
1. As a user
2. I want / need to be able to find available desks for a certain day and time considering my preferences
3. So that I can book my favorite one and work productively at the office that day

## Acceptance criteria
* Function gets parameters from UI page, such as date, time and optional preferences (country, location, desk properties, room etc.)
* Function should identify and return possible free desk options in a sorted manner (country -> location -> room etc.) so that they can be efficiently displayed

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
